src/Microsoft.AspNetCore.Html.Abstractions/HtmlFormattableString.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text.Encodings.Web;
+
+namespace Microsoft.AspNetCore.Html
+{
+    /// <summary>
+    /// An <see cref="IHtmlContent"/> implementation of composite string formatting 
+    /// (see https://msdn.microsoft.com/en-us/library/txafckwd(v=vs.110).aspx) which HTML encodes
+    /// formatted arguments.
+    /// </summary>
+    [DebuggerDisplay("{DebuggerToString()}")]
+    public class HtmlFormattableString : IHtmlContent
+    {
+        private readonly IFormatProvider _formatProvider;
+        private readonly string _format;
+        private readonly object[] _args;
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlFormattableString"/> with the given <paramref name="format"/> and
+        /// <paramref name="args"/>.
+        /// </summary>
+        /// <param name="format">A composite format string.</param>
+        /// <param name="args">An array that contains objects to format.</param>
+        public HtmlFormattableString(string format, params object[] args)
+                : this(null, format, args)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlFormattableString"/> with the given <paramref name="formatProvider"/>,
+        /// <paramref name="format"/> and <paramref name="args"/>.
+        /// </summary>
+        /// <param name="formatProvider">An object that provides culture-specific formatting information.</param>
+        /// <param name="format">A composite format string.</param>
+        /// <param name="args">An array that contains objects to format.</param>
+        public HtmlFormattableString(IFormatProvider formatProvider, string format, params object[] args)
+        {
+            if (format == null)
+            {
+                throw new ArgumentNullException(nameof(format));
+            }
+
+            if (args == null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            _formatProvider = formatProvider ?? CultureInfo.CurrentCulture;
```

Comment:
allowing the format provider to be null is consistent with `string.Format` and other things like this.